File: MyTeletouch.Entities/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MyTeletouch.Entities
{
    public class Product : BaseModel
    {
        [Required(ErrorMessageResourceName = "validation_required_field", ErrorMessageResourceType = typeof(Resources.Resources))]
        [Display(Name = "product_code", ResourceType = typeof(Resources.Resources))]
        [MaxLength(128, ErrorMessageResourceName = "validation_max_field_lenght", ErrorMessageResourceType = typeof(Resources.Resources))]
        public string InternalCode { get; set;  }

        [Display(Name = "product_image", ResourceType = typeof(Resources.Resources))]
        public string ImagePath { get; set; }

        [Display(Name = "unit_price", ResourceType = typeof(Resources.Resources))]
        public double? UnitPrice { get; set; }
    }
}

File: MyTeletouch/Scripts/angular/models/Product.ts
/// <reference path="basemodel.ts" />

module Myteletouch {
    "use strict";

    export module Model {
        export class Product extends BaseModel {
            private _InternalCode: string;
            private _ImagePath: string;
            private _UnitPrice: number;

            constructor() {
                super();

                this._InternalCode = null;
            }

            get Id(): number {
                return this.Id;
            }

            get InternalCode(): string {
                return this._InternalCode;
            }

            set InternalCode(newValue: string) {
                this._InternalCode = newValue;
            }

            get ImagePath(): string {
                return this._InternalCode;
            }

            set ImagePath(newValue: string) {
                this._ImagePath = newValue;
            }

            get UnitPrice(): number {
                return this._UnitPrice;
            }

            set UnitPrice(newValue: number) {
                this._UnitPrice = newValue;
            }
        }
    }
}